Develop

meshack-mbuvi opened the pull request:
In this pull request, the following have been implemented and ready for release:
- user account creation
- user login and logout
- user password reset
- creation of a new book
- update of book information
- book borrowing
- book deletion
- listing all books
- Retrieving a particular book
![register](https://user-images.githubusercontent.com/9263906/38249432-4c5eff76-3754-11e8-8c92-36ace02f595c.png)
![reset](https://user-images.githubusercontent.com/9263906/38249465-6699af08-3754-11e8-8534-01a512ce0dbf.png)
![newbook](https://user-images.githubusercontent.com/9263906/38249470-6d8fb848-3754-11e8-965a-1cb655689781.png)
![deletebook](https://user-images.githubusercontent.com/9263906/38249483-78f21ed8-3754-11e8-8297-d7cfef86a1fd.png)
![deletebook](https://user-images.githubusercontent.com/9263906/38249494-7d7911be-3754-11e8-83ad-f8f562316cb2.png)
![onebook](https://user-images.githubusercontent.com/9263906/38249496-81f2404e-3754-11e8-925b-bca20a971bc5.png)

Commit:
Fix failing tests

Commit:
[Feature #156303248] Add test case for creating user with empty fields.

Add a test case for using an existing username when creating a new account.

Commit:
Edit returned message when empty fields are used to create new account.

Commit:
Remove unnecessary files for this feature like tests for books

Edit the message returned in method post when creating new user

meshack-mbuvi committed:
Merge pull request #24 from meshack-mbuvi/ft-user-login-156143866

Ft user login 156143866

Commit:
Merge branch 'develop' of https://github.com/meshack-mbuvi/hello-books-api into develop

Commit:
Fix failing tests for chaning user password

Commit:
[Feature #156417996] Add test cases for user logout

Commit:
Add a list for holding blacklisted tokens

Commit:
[Feature #156417996] Add function for logging out users

Commit:
[Feature #156417996] Add more configurations used when logging out users

 [Finishes #156417996]

Commit:
Refactor code to remove code segments which are duplicates.

meshack-mbuvi committed:
Merge branch 'develop' into ft-user-log-out-156417996

Commit:
Refactor code further by using short circuiting when checking for bad format and empty fields.

meshack-mbuvi committed:
Merge pull request #26 from meshack-mbuvi/ft-user-log-out-156417996

Ft user log out 156417996

Xerrex commented:
Be consistent with your commit messages. ie they contain the Id of the story your working on.

Commit:
[Feature #156306687] Add dynamic variable in the url for borrowing a book

Commit:
[Feature #156306687] Modify test cases to reflect the changes made on the url for borrowing a book

Commit:
[Feature #156479106] Add login in setup function in testcases and populate users_table with dummy users.

Commit:
[Feature #156479106] Add user_id to Book model.

Commit:
[Feature #156479106] Modify method for borrowing a book so as it modifies the field on the book.

Commit:
Remove books_record; it is no longer needed.

Commit:
Merge branch 'develop' into ft-borrow-book-156479106

Commit:
[Finishes #156479106] Add attribute available to book models

Commit:
[Finishes #156479106]
- Change borrow book endpoint so that when user borrows a book, his/her user_id is added to book data.
- Fix failing tests resulting from these modifications .
- Correct the status code on auth/view.py when user fails to create an account due to used username.

Commit:
Modify base url in setup method for all test classes since testclient() creates its own server

Commit:
Change variable names to be descriptive

Commit:
Improve code % coverage by refactoring and creating more tests.

Commit:
Created a new file to fit purpose.

meshack-mbuvi committed:
Merge pull request #29 from meshack-mbuvi/ft-book-borrowing-156306687

Ft book borrowing 156306687

Commit:
Merge branch 'ft-borrow-book-156479106' into develop

Commit:
Merge branch 'develop' of https://github.com/meshack-mbuvi/hello-books-api into develop

Commit:
Revert to working copy

Commit:
Fix errors introduced by forward slashes.

Commit:
Delete database.py
